index.js

```diff
@@ -42,3 +27,9 @@ export default function nanoSpawn(command, arguments_ = [], {signal, timeout, na
 		/* eslint-enable promise/prefer-await-to-then */
 	};
 }
+
+// TODO: Pass in `stdin` and `stdout` strings here.
+const waitForExit = async subprocess => {
+	const [exitCode, signalName] = await once(subprocess, 'close');
+	return {exitCode, signalName};
```

Comment:
I double checked the core Node.js code, and just listening for the `close` and `error` events catches every possible exit situation. :+1: 

Also, unlike the `exit` event, the `close` event waits for IPC and file descriptors to close.